lib/endpoints/class-wp-rest-users-controller.php

```diff
@@ -841,6 +858,51 @@ public function get_collection_params() {
 			'type'               => 'string',
 			'validate_callback'  => 'rest_validate_request_arg',
 		);
+		$query_params['roles']   = array(
+			'description'        => __( 'Limit result set to resources matching at least one specific role provided. Accepts csv list or single role.' ),
+			'type'               => 'array',
+			'sanitize_callback'  => array( $this, 'rest_sanitize_user_roles' ),
+			'validate_callback'  => array( $this, 'rest_validate_user_roles' ),
+		);
 		return $query_params;
 	}
+
+	/**
+	 * Return a sanitized array of user roles.
+	 *
+	 * @param  mixed $roles Either an array of roles or an array
+	 * @return array $roles Sanitized user roles.
+	 */
+	public function rest_sanitize_user_roles( $roles ) {
```

Comment:
Instead of introducing a method on this class, I want to create a `wp_parse_slug_list` to handle parsing a list of slugs. See https://core.trac.wordpress.org/ticket/35582 for the patch.

We can include our copy of the function in `core-integration.php`